src/mlpack/tests/sarah_test.cpp

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file sarah_test.cpp
+ * @author Marcus Edel
+ *
+ * Test file for SARAH and SARAH+.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/optimizers/sarah/sarah.hpp>
+
+#include <mlpack/methods/logistic_regression/logistic_regression.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace std;
+using namespace arma;
+using namespace mlpack;
+using namespace mlpack::optimization;
+
+using namespace mlpack::distribution;
+using namespace mlpack::regression;
+
+BOOST_AUTO_TEST_SUITE(SAHRATest);
+
+/**
+ * Create the data for the logistic regression test case.
+ */
+void CreateLogisticRegressionTestData(arma::mat& data,
+                                      arma::mat& testData,
+                                      arma::mat& shuffledData,
+                                      arma::Row<size_t>& responses,
+                                      arma::Row<size_t>& testResponses,
+                                      arma::Row<size_t>& shuffledResponses)
```

Comment:
We could add this to an auxiliary header, because I think it's the same across all three tests (and also used in some other places); what do you think?